lib/miq_concurrency.rb

```diff
@@ -0,0 +1,53 @@
+module MiqConcurrency
+  module PGMutex
+    class << self
+      require 'zlib'
+      require 'securerandom'
+
+      def try_advisory_lock(lock_name)
+        establish_connection()
+        hashcode = stable_hashcode(lock_name)
+        execute_successful?('pg_try_advisory_lock', hashcode)
```

Comment:
This seems more indirection than is really required: https://github.com/rails/rails/blob/ce33443d729fb3f2a5a933b6f0e848f17b2c0751/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L313-L325